### DISCUSSION TITLE
需要快速使用指南

### DISCUSSION THREAD

**Discussion by fastfading** [AGou-ops/dotfiles#10] [Q&A]
快捷键
功能说明
等等，
插件虽好，  不知道怎么使用啊

**Comment by fastfading** [comment]
比如 
outline
floatterm 
goto definiton  / reference 
golang 的一些操作 怎么用
全目录搜索
文本替换

等等。。。

**Comment by fastfading** [comment]
很多功能不work
需要安装额外包

,fg
rg: Executable not found

,fF
ranger ： Executable not found


vim with python3 



LeaderF  这个好难装， 卡在这了 好像还要和rg 联动

Error detected while processing /Users/huzhang2/.vim/plugged/LeaderF/autoload/leaderf/Any.vim[10]../Users/huzhang2/.vim/plugged/LeaderF/autoload/leaderf.vim
:
line   18:
Error: LeaderF requires vim compiled with +python or +python3
Error: LeaderF requires vim compiled with +python or +python3
Error detected while processing /Users/huzhang2/.vim/plugged/LeaderF/autoload/leaderf/Any.vim:
line   10:
E117: Unknown function: leaderf#versionCheck
line   14:
E121: Undefined variable: g:Lf_py
Error detected while processing function leaderf#Any#start[4]..script /Users/huzhang2/.vim/plugged/LeaderF/autoload/leaderf.vim:
line   18:
Error: LeaderF requires vim compiled with +python or +python3
Error detected while processing function leaderf#Any#start:
line    4:
E117: Unknown function: leaderf#LfPy

**Comment by fastfading** [comment]
sf
golang   reference  不能跳转到文件，

**Comment by AGou-ops** [comment]
> 很多功能不work 需要安装额外包
> 
> ,fg rg: Executable not found
> 
> ,fF ranger ： Executable not found
> 
> vim with python3
> 
> LeaderF 这个好难装， 卡在这了 好像还要和rg 联动
> 
> Error detected while processing /Users/huzhang2/.vim/plugged/LeaderF/autoload/leaderf/Any.vim[10]../Users/huzhang2/.vim/plugged/LeaderF/autoload/leaderf.vim : line 18: Error: LeaderF requires vim compiled with +python or +python3 Error: LeaderF requires vim compiled with +python or +python3 Error detected while processing /Users/huzhang2/.vim/plugged/LeaderF/autoload/leaderf/Any.vim: line 10: E117: Unknown function: leaderf#versionCheck line 14: E121: Undefined variable: g:Lf_py Error detected while processing function leaderf#Any#start[4]..script /Users/huzhang2/.vim/plugged/LeaderF/autoload/leaderf.vim: line 18: Error: LeaderF requires vim compiled with +python or +python3 Error detected while processing function leaderf#Any#start: line 4: E117: Unknown function: leaderf#LfPy

```bash
pip2 install --user --upgrade neovim
pip3 install --user --upgrade neovim
```

leadF的依赖问题， 安装Python2 和 Python3 支持就行了.

`rg`和`ranger`的问题，直接安装一下就行了.

```bash
apt install rg ranger
// macos
brew install rg ranger
```

**Comment by AGou-ops** [comment]
> 比如 outline floatterm goto definiton / reference golang 的一些操作 怎么用 全目录搜索 文本替换
> 
> 等等。。。

这个在`lua/AGou/lsp.lua`文件里面有写，都是快捷键，看英文就行了.
![image](https://user-images.githubusercontent.com/57939102/202903030-7293f545-5b84-402a-9a17-f45d37a3f1ff.png)

**Comment by AGou-ops** [comment]
> 比如 outline floatterm goto definiton / reference golang 的一些操作 怎么用 全目录搜索 文本替换
> 
> 等等。。。

这些你看`keymap.vim`这个文件就行了，快捷键都在这里面，我都有注释是哪个插件的快捷键.

实在是没空写那么详细了.😂

**Comment by fastfading** [comment]
<img width="935" alt="image" src="https://user-images.githubusercontent.com/8351732/202972367-c1ea80a4-a33b-48c7-aef5-b71f518fd8e8.png">
这个快捷键是什么

**Comment by AGou-ops** [comment]
> <img alt="image" width="935" src="https://user-images.githubusercontent.com/8351732/202972367-c1ea80a4-a33b-48c7-aef5-b71f518fd8e8.png">
> 
> 这个快捷键是什么

那个是关闭buffer，鼠标可以直接点击，关闭当前buffer的快捷键是`C-q`.

**Comment by fastfading** [comment]
> > <img alt="image" width="935" src="https://user-images.githubusercontent.com/8351732/202972367-c1ea80a4-a33b-48c7-aef5-b71f518fd8e8.png">
> > 这个快捷键是什么
> 
> 那个是关闭buffer，鼠标可以直接点击，关闭当前buffer的快捷键是`C-q`.

鼠标点x  和ctrl+q       :bd         <leader>dd  行为是不一样的

只有 鼠标点x  能完成完美关闭动作，并跳到上一个没有关闭的页面，  这个和ide 行为基本一致
  其他会吧窗口关掉，然后留出一片空白。

**Comment by AGou-ops** [comment]
> > > <img alt="image" width="935" src="https://user-images.githubusercontent.com/8351732/202972367-c1ea80a4-a33b-48c7-aef5-b71f518fd8e8.png">
> > > 这个快捷键是什么
> > 
> > 
> > 那个是关闭buffer，鼠标可以直接点击，关闭当前buffer的快捷键是`C-q`.
> 
> 鼠标点x 和ctrl+q :bd dd 行为是不一样的
> 
> 只有 鼠标点x 能完成完美关闭动作，并跳到上一个没有关闭的页面， 这个和ide 行为基本一致 其他会吧窗口关掉，然后留出一片空白。

`C-q`我是映射的`C-w q`，这个快捷键是关闭窗口，比如分屏的时候关闭当前窗口保留另外一个，点`x`号用的是bufdelete，会保留当前的window，也就是你上面说的.  我一般单窗口多buffer鼠标点击`x`号或者使用`:bd `来删除buffer，多窗口分屏的时候用`C-q`.

具体是在下面这两个地方设置的：

> https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/keymaps.vim#L232

> https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/lua/AGou/nvim-bufferline.lua#L6-L8

**Comment by fastfading** [comment]
> > > > <img alt="image" width="935" src="https://user-images.githubusercontent.com/8351732/202972367-c1ea80a4-a33b-48c7-aef5-b71f518fd8e8.png">
> > > > 这个快捷键是什么
> > > 
> > > 
> > > 那个是关闭buffer，鼠标可以直接点击，关闭当前buffer的快捷键是`C-q`.
> > 
> > 
> > 鼠标点x 和ctrl+q :bd dd 行为是不一样的
> > 只有 鼠标点x 能完成完美关闭动作，并跳到上一个没有关闭的页面， 这个和ide 行为基本一致 其他会吧窗口关掉，然后留出一片空白。
> 
> `C-q`我是映射的`C-w q`，这个快捷键是关闭窗口，比如分屏的时候关闭当前窗口保留另外一个，点`x`号用的是bufdelete，会保留当前的window，也就是你上面说的. 我一般单窗口多buffer鼠标点击`x`号或者使用`:bd `来删除buffer，多窗口分屏的时候用`C-q`.
> 
> 具体是在下面这两个地方设置的：
> 
> > https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/keymaps.vim#L232
> 
> > https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/lua/AGou/nvim-bufferline.lua#L6-L8

要用  :Bdelete  
bdelete 和 Bdelete  还不一样

**Comment by fastfading** [comment]
另外 ctrl-i  的行为被篡改了， 我希望是往前跳， 和vim 一样。     
    我改了 keymaps 也不行

**Comment by AGou-ops** [comment]
> > > > > <img alt="image" width="935" src="https://user-images.githubusercontent.com/8351732/202972367-c1ea80a4-a33b-48c7-aef5-b71f518fd8e8.png">
> > > > > 这个快捷键是什么
> > > > 
> > > > 
> > > > 那个是关闭buffer，鼠标可以直接点击，关闭当前buffer的快捷键是`C-q`.
> > > 
> > > 
> > > 鼠标点x 和ctrl+q :bd dd 行为是不一样的
> > > 只有 鼠标点x 能完成完美关闭动作，并跳到上一个没有关闭的页面， 这个和ide 行为基本一致 其他会吧窗口关掉，然后留出一片空白。
> > 
> > 
> > `C-q`我是映射的`C-w q`，这个快捷键是关闭窗口，比如分屏的时候关闭当前窗口保留另外一个，点`x`号用的是bufdelete，会保留当前的window，也就是你上面说的. 我一般单窗口多buffer鼠标点击`x`号或者使用`:bd `来删除buffer，多窗口分屏的时候用`C-q`.
> > 具体是在下面这两个地方设置的：
> > > https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/keymaps.vim#L232
> > 
> > 
> > > https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/lua/AGou/nvim-bufferline.lua#L6-L8
> 
> 要用 :Bdelete bdelete 和 Bdelete 还不一样

...

我用的不就是你说的那个`Bdelete`？`Bdelete`是[这个插件](https://github.com/famiu/bufdelete.nvim)提供的，我用require调用了.

https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/lua/AGou/nvim-bufferline.lua#L7

```bash
require('bufdelete').bufdelete(bufnum, true)
```

**Comment by AGou-ops** [comment]
> 另外 ctrl-i 的行为被篡改了， 我希望是往前跳， 和vim 一样。 我改了 keymaps 也不行

快捷键都是根据自己的习惯改的，不喜欢或者不适合你可以直接改，我上面也说了，都在`keymap.vim`和`lsp.lua`这两个文件里面.

你说的`C-i`不就在这里？

https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/keymaps.vim#L49


改了没生效那可能是你没重载配置文件.

**Reply by fastfading** [reply]
还需要改 tab  映射 %  

golang  outline   怎么打开

**Reply by AGou-ops** [reply]
tab你可以自己改，我用来快速跳转括号的.

outline 我用的是`tagbar`插件，需要[安装依赖](https://github.com/preservim/tagbar#dependencies)，这个打开的方法是在nvim命令行模式输入`:Tagbar`就行了，如果不喜欢tagbar，你可以换成别的插件.

<img width="1534" alt="image" src="https://user-images.githubusercontent.com/57939102/203206839-e3cce0c1-fcc9-4f46-886a-9d3e17c9cf2d.png">

**Comment by fastfading** [comment]
nnoremap gR <cmd>TroubleToggle lsp_references<cr>

gR   第一次的时候没有任何问题，  
enter 跳转到相应代码后，  trouble window 还在。 
gR   第二次，  trouble window 就关闭了。这个体验非常不好，  该如何改进

**Reply by AGou-ops** [reply]
`TroubleToggle`(另外还有`Trouble`和`TroubleClose`)的意思是触发关闭或者开启Trouble，是一个全局的，你第一次跳转打开了Trouble，再用一次快捷键自然就关闭掉了，你可以使用`autocmd`自动打开，这个你自己谷歌学学吧，还有一个是手动打开或者关闭.

**Comment by fastfading** [comment]
多谢您耐心的解答，  
我找到一个配置更简单，帮助文档更多的nvim     lunarvim  。  

最后一个问题：   我看您配置的配色方案特别好看，   这个配色方案如何安装啊。
https://github.com/sainnhe/gruvbox-material

**Reply by AGou-ops** [reply]
主题仓库的文档里面有教程啊.

https://github.com/sainnhe/gruvbox-material/blob/master/doc/gruvbox-material.txt#L83

**Reply by fastfading** [reply]
```
" For dark version.
set background=dark
" Set contrast.
" This configuration option should be placed before `colorscheme gruvbox-material`.
" Available values: 'hard', 'medium'(default), 'soft'
let g:gruvbox_material_background = 'hard'
" For better performance
let g:gruvbox_material_better_performance = 1

let g:gruvbox_material_diagnostic_text_highlight = 1
" let g:gruvbox_material_diagnostic_line_highlight = 1
let g:gruvbox_material_diagnostic_virtual_text = "colored"
let g:gruvbox_material_sign_column_background = 'none'

colorscheme gruvbox-material
```

我看你color配置写在vim 文件里了，
要写在lua 里， 该怎么写

**Reply by AGou-ops** [reply]
这个主题没有lua相关的配置，只能用viml来写，你可以用`source xx.vim`在lua中引用viml配置.

**Comment by fastfading** [comment]
看来看去，还是你的这个配置好。  
稍为做了一些 keymap 改动就很好用了

另外  comment  快捷键是什么， 我没有找到

**Reply by AGou-ops** [reply]
快捷键，基本全都在`keymap.vim`和`lua/AGou/lsp.lua`这两个文件里面.

https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/lua/AGou/init.lua#L46-L53

注释的快捷键这个地方有写：

```lua
-- full hotkeys usage: https://github.com/numToStr/Comment.nvim#-usage
-- gcc: 当前行行注释
-- gcb: 当前行块注释
-- [count]gcc/gcb: 多行块或行注释
-- `gco` - Insert comment to the next line and enters INSERT mode
-- `gcO` - Insert comment to the previous line and enters INSERT mode
-- `gcA` - Insert comment to end of the current line and enters INSERT mode
require("Comment").setup({})

```

**Comment by fastfading** [comment]
另外    live_grep  能不能换为 live_grep_args  , 可以指定搜索pattern    -g --glob
lvim.keys.normal_mode["<leader>fg"] = ":lua require('telescope').extensions.live_grep_args.live_grep_args()<CR>"

**Reply by AGou-ops** [reply]
https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/keymaps.vim#L155-L165

在这里改就行了，加参数直接加就可以.

**Comment by fastfading** [comment]
还有ctrl d   ctrl u 为什么这么顺滑， 用的什么插件啊

**Reply by AGou-ops** [reply]
`neoscroll`这个插件.

**Comment by fastfading** [comment]
<img width="990" alt="image" src="https://user-images.githubusercontent.com/8351732/206066347-34396737-7906-41c0-940b-25b87ee84681.png">
这是怎么回事

**Reply by AGou-ops** [reply]
这是我k8s yaml文件lsp读取的配置文件，你不需要的话直接删掉吧.

在这里：

https://github.com/AGou-ops/dotfiles/blob/master/neovim-with-lua/lua/AGou/lsp.lua#L114
